addons/mass_mailing/static/src/snippets/s_social_media_mass_mailing/options.js

```diff
@@ -0,0 +1,109 @@
+odoo.define('@web_editor/src/snippets/s_social_media_mass_mailing/options', function (require) {
+
+const options = require('web_editor.snippets.options');
+const {SocialMedia} = require('@web_editor/js/editor/snippets/s_social_media/options');
+const rpc = require('web.rpc');
+const session = require('web.session');
+
+let dbSocialValues;
+let dbSocialValuesProm;
+options.registry.SocialMediaMassMailing = SocialMedia.extend({
+    /**
+     * @override
+     */
+    async cleanForSave() {
+        await this._fetchSocialMedia();
+        const dbSocialValuesToSave = new Set();
+        for (const [mediaName, url] of Object.entries(dbSocialValues)) {
+            if (!url) {
+                dbSocialValuesToSave.add(mediaName);
+            }
+        }
+        const dict = {};
+        for (const entry of this._lastEntries) {
+            if (entry.media && entry.display_name && dbSocialValuesToSave.has(entry.media)) {
+                dbSocialValues[entry.media] = entry.display_name;
+                dict[`social_${entry.media}`] = entry.display_name;
+                dbSocialValuesToSave.delete(entry.media);
+            }
+        }
+
+        const query = rpc.buildQuery({
+            model: 'res.company',
+            method: 'write',
+            args: [[session.company_id], dict],
+        });
+        await session.rpc(query.route, query.params).catch(e => {
+            if (e.message.exceptionName !== 'odoo.exceptions.AccessError') {
+                return Promise.reject(e);
+            }
+        });
```

Comment:
- It feels scary to not call `_super` in a `cleanForSave` extension of an existing class. The purpose here seems to be the same as the original class: save the values in db. Can't there be an abstract method in common class (called something like `_saveSocialMediaValues`), called in cleanForSave of the common class, and then implemented by the website specific class and the mass mailing specific class?
- Can't you do `this._rpc` instaed of that `buildQuery` and `session.rpc` stuff ?